fppanalysis/two_dim_velocity_estimates.py

```diff
@@ -75,21 +75,49 @@ def _get_dt(ds):
         return times[1].values - times[0].values
     raise "Unknown format"
 
+def _get_time(x, y, ds):
+    # Sajidah's format
+    if hasattr(ds, "time"):
+        return ds.isel(x=x, y=y).time.values
+    # 2d code
+    if hasattr(ds, "t"):
+        return ds.isel(x=x, y=y).time.values
+    raise "Unknown format"
+
 
-def _estimate_velocities_given_points(p0, p1, p2, ds):
+def _estimate_velocities_given_points(p0, p1, p2, ds, cross_cond_av: bool):
+    """Estimates radial and poloidal velocity from estimated time delay 
+    either from cross conditional average (if cross_cond_av = True)
+    between the pixels or cross correlation (if cross_cond_av = False). 
+    In cross conditional average, time array is required in order to normalize
+    the signal before averageing.  
+    """
     dt = _get_dt(ds)
     r0, z0 = _get_rz(p0[0], p0[1], ds)
     r1, z1 = _get_rz(p1[0], p1[1], ds)
     r2, z2 = _get_rz(p2[0], p2[1], ds)
     signal0 = _get_signal(p0[0], p0[1], ds)
     signal1 = _get_signal(p1[0], p1[1], ds)
     signal2 = _get_signal(p2[0], p2[1], ds)
+    time0 = _get_time(p0[0], p0[1], ds)
+    time1 = _get_time(p1[0], p1[1], ds)
+    time2 = _get_time(p2[0], p2[1], ds)
 
     if len(signal0) == 0 or len(signal1) == 0 or len(signal2) == 0:
         return None
-
-    delta_ty, cy = tde.estimate_time_delay_ccmax(x=signal2, y=signal0, dt=dt)
-    delta_tx, cx = tde.estimate_time_delay_ccmax(x=signal1, y=signal0, dt=dt)
+    
+    if cross_cond_av:
+        delta_tx, cond_variance_x, events_x = tde.estimate_time_delay_ccond_av_max(signal1, time1, signal0, time0)
+        delta_ty, cond_variance_y, events_y = tde.estimate_time_delay_ccond_av_max(signal2, time2, signal0, time0)
+        
+        # Confidence when velocities are estimated from 
+        # cross conditional averge is: 1 - conditional variance
+        cx = 1 - cond_variance_x
+        cy = 1 - cond_variance_y
+    else: 
+        delta_ty, cy = tde.estimate_time_delay_ccmax(x=signal2, y=signal0, dt=dt)
+        delta_tx, cx = tde.estimate_time_delay_ccmax(x=signal1, y=signal0, dt=dt)
```

Comment:
Even though it might not be 100% necessary, we will increase the complexity of this soon (by having more options on how to compute delta_tx, delta_ty).
In order to improve readability an simplify code write a new function "_estimate_time_delay(x, y, times_x, times_y, method)" that returns delta_t, c by calling the appropriate underlying estimate_time_delay function. That way, this function (_estimate_velocities_given_points) will preserve the same structure and simplicity that it had before.